client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -0,0 +1,369 @@
+			...newState,
+			...( dateOfBirth ? { dateOfBirth } : {} )
+		} );
+	}
+
+	newStateFromEvent( event ) {
+		return {
+			...this.props.values,
+			[ event.target.id ]: event.target.value,
+		};
+	}
+
+	handleChangeEvent = ( event ) => {
+		this.props.onStateChange( this.newStateFromEvent( event ) );
+	}
+
+	// We need a deep comparison to check inside props.values
+	shouldComponentUpdate = ( nextProps ) => {
+		return ! isEqual( this.props, nextProps );
+	}
+
+	render = () => {
+		const translate = this.props.translate;
+		const {
+			registrantType
+		} = { ...emptyValues, ...this.props.values };
+
+		return (
+			<form className="registrant-extra-info__form">
+				<h1 className="registrant-extra-info__form-title">
+					{ translate(
+						'Registering a .FR domain'
+					) }
+				</h1>
+				<p className="registrant-extra-info__form-desciption">
+					{ translate(
+						'Almost done! We need some extra details to register domains ending in ".fr".'
+					) }
+				</p>
+				<FormFieldset>
+					<FormLegend>
+						{ translate( "Who's this domain for?" ) }
+					</FormLegend>
+					<FormLabel>
+						<FormRadio value="individual"
+							id="registrantType"
+							checked={ 'individual' === registrantType }
+							onChange={ this.handleChangeEvent } />
+						<span>{ translate( 'An individual' ) }</span>
+					</FormLabel>
+
+					<FormLabel>
+						<FormRadio value="organization"
+							id="registrantType"
+							checked={ 'organization' === registrantType }
+							onChange={ this.handleChangeEvent } />
+						<span>{ translate( 'A company or organization' ) }</span>
+					</FormLabel>
+				</FormFieldset>
+
+				{ 'individual' === registrantType
+					? this.renderPersonalFields()
+					: this.renderOrganizationFields() }
+
+				{ this.props.children }
+			</form>
+		);
+	}
+
+	renderPersonalFields = () => {
+		const translate = this.props.translate;
+		const screenReaderText = 'screen-reader-text';
+		const {
+			countryOfBirth,
+			dobDays,
+			dobMonths,
+			dobYears,
+			placeOfBirth,
+			postalCodeOfBirth,
+		} = { ...emptyValues, ...this.props.values };
+
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel htmlFor="countryOfBirth">
+						{ translate( 'Country of Birth' ) }
+					</FormLabel>
+					<FormCountrySelect
+						id="countryOfBirth"
+						value={ countryOfBirth }
+						countriesList={ this.props.countriesList }
+						className="registrant-extra-info__form-country-select"
+						onChange={ this.handleChangeEvent } />
+				</FormFieldset>
+
+				<FormFieldset>
+					<FormLegend>
+						{ translate( 'Date of Birth' ) }
+					</FormLegend>
+					<div className="registrant-extra-info__dob-inputs">
+						<div className="registrant-extra-info__dob-column">
+							<FormLabel htmlFor="dobYears" className={ screenReaderText }>
+								{ translate( 'Year' ) }
+							</FormLabel>
+							<FormTextInput className="registrant-extra-info__dob-year"
+								id="dobYears"
+								value={ dobYears }
+								max="4"
```

Comment:
I'll just drop the `max` for now, but css validation does seem like bit of a missed opportunity.

I don't know if our custom submit handling is compatible with the validation check, but even if it wasn't we could have the error elements ready and then make them visible for invalid values.

Js validation seems a fair choice, though. Much more flexibility and control.